app/com/lucidchart/open/cashy/views/upload/index.scala.html

```diff
@@ -134,28 +158,266 @@ <h4 class="modal-title" id="myModalLabel">Uploading an Image</h4>
     </div>
   </div>
 
-  <script>
+  @if(krakenEnabled) {
+    <!-- modal for krakened image -->
+    <div class="modal fade" id="imagePreviewModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+      <div class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+            <h4 class="modal-title" id="myModalLabel">Resized Image Preview</h4>
+          </div>
+          <div class="modal-body">
+            Note: Large images will be scaled down.  Click on each image to open it in a new tab.
+            <div id="originalImage">
+              <h2>Original</h2>
+              <a href="" target="_blank"><img src="" /></a>
+            </div>
+            <div id="resizedImage">
+              <h2>Resized</h2>
+              <a href="" target="_blank"><img src="" /></a>
+            </div>
+          </div>
+          <div class="modal-footer" style="clear: left;">
+            <button type="button" class="btn btn-success" id="confirmResizePreview">Resize</button>
+            <button type="button" class="btn btn-danger" id="cancelResizePreview">Don't Resize</button>
+          </div>
+        </div>
+      </div>
+    </div>
+  }
+
+<style>
+  .check-show {
+    display: none;
+  }
+  input[type="checkbox"]:checked~.check-show {
+    display: block;
+  }
+  dl.inline-input {
+    display: inline-block;
+  }
+  #imageUploadInfo dl {
+    margin-bottom: 0;
+  }
+  #uploadRetina {
+    margin-top: 20px;
+  }
+  .resize-error, .retina-error {
+    color: red;
+  }
+  #imageUploadInfo label {
+    font-weight: normal;
+  }
+  #imageUploadInfo dd.info {
+    display: none;
+  }
+  #resizedImage, #originalImage {
+    max-width: 250px;
+    display: inline-block;
+    float: left;
+    margin-bottom: 20px;
+    margin-right: 10px;
+  }
+
+  #resizedImage img, #originalImage img {
+    max-width: 250px;
+  }
+  #previewImageButton:disabled {
+    cursor: wait;
+  }
+</style>
+
+<script>
 
   $('#fileInput').change(function(e) {
-    var assetFullName = $('#fileName').val()
+    var assetFullName = $('#fileName').val();
     var assetFileName = assetFullName.split(/(\\|\/)/g).pop();
     var uploadFileName = $(this).val().split(/(\\|\/)/g).pop();
-    var extension = uploadFileName.split(/\./g).pop();
+
+    if (assetFileName.length == 0) {
+      $('#fileName').val(assetFullName + uploadFileName);
+    } else {
+      $('#fileName').val(assetFullName.replace(assetFileName, uploadFileName));
+    }
+    assetFullName = $('#fileName').val();
+    var extension = assetFullName.split(/\./g).pop();
```

Comment:
Using pop, we take the last extension they specify site.min.js -> js and so on.  This extension var is only used to check if I should show the additional info for js/css/images or in the case of images to create the @2x name for a file.  In the case of images I don't expect them to use multiple extensions, and if they do they can edit the names themselves anyways.